### ISSUE TITLE
Breadrumbs: alignment of text vs icons vs twistie is off

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#67342]
I find that the alignment of breadcrumbs is weird. I am on Linux with zoom level 1 (though I also see it without zooming). To emphasize the box around breadcrumbs I added a green background:

![image](https://user-images.githubusercontent.com/900690/51900275-32368f80-23b5-11e9-9dd8-d01cf0a2d8ad.png)

Somehow the alignment of text, twistie and icon is not really vertically centered.

**Comment by jrieken** [comment]
I don't know why this keep regressing... I didn't change breadcrumbs. @bpasero @isidorn did the `IconLabel` or `ResourceLabel` change?

**Comment by isidorn** [comment]
Did not change the IconLbale or ResourceLabel this milestone. @bpasero can you repro with stable?

**Comment by bpasero** [comment]
@jrieken 

Similar for me on stable:

![image](https://user-images.githubusercontent.com/900690/51906632-d4aa3f00-23c4-11e9-92d1-8ddc2ec70e08.png)

**Comment by jrieken** [comment]
It is just like in the explorer but somehow it looks different


<img width="596" alt="screenshot 2019-01-29 at 13 00 58" src="https://user-images.githubusercontent.com/1794099/51907080-256e6780-23c6-11e9-89ff-c0a73e4d4e79.png">

**Comment by bpasero** [comment]
**Extracting** the breadcrumb to illustrate a few issues I am seeing:

![52058745-656d5000-2568-11e9-95d7-46b66c880512](https://user-images.githubusercontent.com/900690/52059165-59ce5900-2569-11e9-9c2f-8fe7ad21777b.png)

* the expand/collapse twistie seems to be more upwards compared to the explorer and it looks like removing [this 22px height rule](https://github.com/Microsoft/vscode/blob/51765d8/src/vs/workbench/browser/parts/editor/media/tabstitlecontrol.css#L295) changes that for me on Linux at least (though of course this was added to fix the same alignment issues on Windows via https://github.com/Microsoft/vscode/issues/57349 so we might need a platform dependent solution)
* the first segment with an icon (`tests.htm`) is actually the resource label as we use it in the explorer with a fixed margin between icon and label while the other segments (e.g. `html`) are breadcrumb items that put the icon much closer to the label. as such the breadcrumb segments are not looking very consistent to each other

To summarise I think:
* each segment should use the same spacing, sizing and alignment of twistie to icon and icon to label
* overall the position of label, twistie and icon does not seem well centered (even though I am seeing a similar issue in other trees)

Maybe we need a custom twistie icon to make it look right in the breadcrumb. One challenge for the file/symbol icons is that these can be contributed via extensions as theming option.

@misolori maybe you could design something up that we can then implement or if you want to test your CSS skills on it, chime in?

**Comment by misolori** [comment]
@bpasero I can design it so we have proper sizing and then have someone implement (tried tweaking CSS but didn't get too far with all of the nested styles)

**Comment by jrieken** [comment]
The big challenge is that the icons, depending on the theme, are icon fonts or SVGs

**Comment by misolori** [comment]
@jrieken do the breadcrumbs not show the file icons themes? I noticed that the twisites don't update (compare explorer to breadcrumbs):

<img width="1008" alt="image" src="https://user-images.githubusercontent.com/35271042/52076455-8ff8e200-2543-11e9-8810-f33a1c3d5f3d.png">

**Comment by jrieken** [comment]
For files and folders it uses them icons but the twisty is always there - would be cool to align that with the tree

**Comment by bpasero** [comment]
@misolori good start, though we have to look at it on all platforms and I just did with mixed results:

**macOS**
I am not seeing any change in vertical alignment. Only the spacing between breadcrumbs seems to have changed.

Stable
<img width="524" alt="screenshot 2019-02-05 at 10 43 16" src="https://user-images.githubusercontent.com/900690/52264866-f3fc1b80-2932-11e9-849d-a427238b1a46.png">

Yours
<img width="558" alt="screenshot 2019-02-05 at 10 43 37" src="https://user-images.githubusercontent.com/900690/52264877-fb232980-2932-11e9-958c-fdaffe3764f9.png">

I think the symbol icons have a great alignment compared to the text. The only outlier to me is the file icon which appears quite small. Also, the twistie seems to be ever so slightly above the center line of the text.

**Windows**
Again, vertical alignment seems to be the same:

Stable
![image](https://user-images.githubusercontent.com/900690/52265016-60771a80-2933-11e9-9ba6-680b0f929ff0.png)

Yours
![image](https://user-images.githubusercontent.com/900690/52265030-6836bf00-2933-11e9-9aad-bce216cf685c.png)

**Linux**
Again, vertical alignment seems to be the same:

Stable
![image](https://user-images.githubusercontent.com/900690/52265136-aa600080-2933-11e9-82f6-8a86fdc8405a.png)

Yours
![image](https://user-images.githubusercontent.com/900690/52265202-d1b6cd80-2933-11e9-882c-4f10cc891059.png)

I feel like consistent spacing is right, however now the spacing between twistie and icon seems rather large?

**Comment by bpasero** [comment]
I just realize that the breadcrumb looks a lot better on my MacBook screen (where I took the screenshots from - without zooming) compared to my external display with zoom 1. So we might not be lucky fixing the vertical alignment problems easily...

**Comment by misolori** [comment]
@bpasero thanks for testing, I'll try again and verify on the other platforms + screen densities.

**Comment by bpasero** [comment]
Now that this is enabled by default, we really gotta fix it. Got reminded again by looking at the release notes:

![image](https://user-images.githubusercontent.com/900690/58687933-587c3400-8383-11e9-9269-436d43709154.png)

**Comment by jrieken** [comment]
re https://github.com/microsoft/vscode/issues/67342#issuecomment-460576021 - do we know how much is due to zooming and how many people use zooming?

**Comment by bpasero** [comment]
Btw I recently changed the status bar to exclusively use octicon font for that reason that some things in there was just not properly aligned with text. We probably talked about this before but maybe the best solution is to switch to a font for everything in the breadcrumb, or at least the twistie. We can easily add new icons to our octicon set as needed.

**Comment by jrieken** [comment]
This one would be closest: https://octicons.github.com/icon/chevron-right/ tho doesn't align with the tree (default)

**Comment by bpasero** [comment]
Yeah makes me wonder if we should tackle this for the tree then as well.

**Comment by misolori** [comment]
Yea, we can always add our tree icons to Octicons since we control it now. However, I don't think we can use Octicons in the tree due to Monaco? At least that's the last that I heard, unless Monaco ships with Octicons.

**Comment by jrieken** [comment]
> At least that's the last that I heard, unless Monaco ships with Octicons.

It doesn't but I actually don't know why that is... cc @rebornix

**Comment by lllopo** [comment]
The latest nightly uses a new icon, but the problem persists (and is the same as the vertical content alignment in the tabs). Besides, with the new icons, the chevron icon should be smaller and with a bit more spacing both sides, I think. The worst looking thing remains the vertical alignment, though.

I see people are wondering how this can be fixed. Simple - you make all elements 100% high and align them exclusively through the flexbox. Then a small vertical correction (move up about .153em) is applied to the text elements, because of the letters hangouts below. It woks as I tried it - both for the tabs and breadcrumbs. Looks pretty fine (results with tabs here : [https://github.com/microsoft/vscode/issues/8017#issuecomment-509575698])

Ow, also the breadcrumbs bar needs a few pixel higher (like 4px), I think as it is too tiny now.

**Comment by misolori** [comment]
@lllopo feel free to create a PR, note that we use the system font for macOS/Win/Linux so character spacing will vary.

**Comment by lllopo** [comment]
@misolori Well, I don't feel fully confident that I won't break something somewhere else with the changes and I don't have the time to do a through test. Otherwise I would gladly do it. Still, I might do it :-).
